src/simulator_core/solver/network/assets/base_node_item.py

```diff
@@ -24,13 +23,15 @@
 class BaseNodeItem(ABC):
     """A base class for node items in a network."""
 
-    def __init__(self, number_of_unknowns: int, name: uuid.UUID):
+    def __init__(self, name: str, identifier: str, number_of_unknowns: int):
         """Initializes the BaseNodeItem object with the given parameters.
 
         :param int number_of_unknowns: The number of unknown variables for the item.
-        :param uuid.UUID name: The unique identifier of the item.
+        :param str name: The name of the node.
+        :param str identifier: The unique identifier of the node.
         """
         self.name = name
+        self.id = identifier
```

Comment:
this is already id, the rest why identifier?

Reply:
changed everything to id or _id